From 4ac4b99cd6527256bfa12dfc516cf51d88bfec05 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Tue, 6 Aug 2019 12:37:23 -0500
Subject: [PATCH 1/4] Properly detect a system GTest

---
 CMakeLists.txt      | 6 +++++-
 core/CMakeLists.txt | 6 +++---
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cdd5367..3da4dc7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,9 +70,13 @@ if (BUILD_TESTS AND NOT USE_SYSTEM_GTEST)
     # Include googletest headers.
     include_directories("${gtest_SOURCE_DIR}/include")
 
+	# Alias the targets to their namespaced versions.
+	add_library(GTest::GTest ALIAS gtest)
+	add_library(GTest::Main ALIAS gtest_main)
+
 elseif (BUILD_TESTS AND USE_SYSTEM_GTEST)
 	enable_testing()
-	add_subdirectory(/usr/src/googletest ${GLOBAL_OUTPUT_PATH}/googletest-build)
+	find_package(GTest REQUIRED)
 endif()
 
 # Compiler flags.
diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index e877015..ddedf7d 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -1,6 +1,6 @@
 # libjsonnet
 
-# Remember to update Bazel and Makefile builds when updating this list! 
+# Remember to update Bazel and Makefile builds when updating this list!
 set(LIBJSONNET_HEADERS
     ast.h
     desugarer.h
@@ -60,9 +60,9 @@ function(add_test_executable test_name)
     endif()
     add_executable(${test_name} ${test_name}.${TEST_EXT})
 
-    add_dependencies(${test_name} libjsonnet_static gtest_main)
+    add_dependencies(${test_name} libjsonnet_static GTest::Main)
     target_link_libraries(
-        ${test_name} gtest gtest_main libjsonnet_static)
+        ${test_name} GTest::GTest GTest::Main libjsonnet_static)
 endfunction()
 
 if (BUILD_TESTS)
-- 
2.21.0

